### DISCUSSION TITLE
Ignoring specific fields in a manifest during reconciliation

### DISCUSSION THREAD

**Discussion by au2001** [fluxcd/flux2#3688] [Proposals]
Hi,

I have a cluster running Flux where I am deploying resources for which part of their configuration changes dynamically through other tools.
I would like to ignore those specific changes during reconciliation, but continue to manage the rest of the manifest's fields with Flux.

An example use-case I have for this is when using [KubeVirt](https://kubevirt.io/).
Some VMs' `running` field should be set by Flux at initial apply, but should then be ignored so that it does not shut down/start up at unexpected times.
Furthermore, some resources may change at runtime. For example, new persistent volumes can be mounted during backups, and RAM/CPU can be increased during intensive tasks.
These changes are short-term and the "usual" configuration should still be defined in Flux to be applied at initial deployment, but should not be reverted if it were to change in-cluster.
Other unchanging fields (node selectors...) should also continue being managed by Flux.

According to [the docs](https://fluxcd.io/flux/components/kustomize/kustomization/#reconciliation), I currently have the choice between:
1. Deploying the resources outside of Flux
2. Labelling the resources with `kustomize.toolkit.fluxcd.io/reconcile: disabled`
3. Excluding the dynamic fields from Flux's source and adding `kustomize.toolkit.fluxcd.io/ssa: merge`

But none of these solutions allow to deploy initial values AND continue to manage other fields at the same time.
Is my understanding correct? Or is there a way to achieve what I am looking for?

Otherwise, is it because this goes against some of the core principles of Flux?

If this is a proposal which could be considered, I would be glad to help with its implementation.

**Comment by Eforen** [comment]
I too am having a similer problem with MetalLB

**Reply by thejimnicholson** [reply]
When I install metallb via kustomization, Flux generates an event every time the flux-system kustomization reconciles. What I see on the event is **CustomResourceDefinition/bgppeers.metallb.io configured**. My kustomization is including the metallb manifest like this:

`- github.com/metallb/metallb/config/native?ref=v0.14.5`
